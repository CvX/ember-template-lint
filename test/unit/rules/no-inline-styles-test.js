'use strict';

const generateRuleTests = require('../../helpers/rule-test-harness');

function mapBadStyleExampleToTestRule(inlineStyle) {
  return {
    template: `<div ${inlineStyle}></div>`,

    verifyResults(results) {
      expect(results).toMatchSnapshot();
    },
  };
}

function mapBadStyleExampleToTestRuleWithConfig(inlineStyle, config) {
  const testCase = mapBadStyleExampleToTestRule(inlineStyle);
  testCase.config = config;
  return testCase;
}

generateRuleTests({
  name: 'no-inline-styles',

  config: true,

  good: [
    '<div></div>',
    '<span></span>',
    '<ul class="dummy"></ul>',
    '<div style={{foo}}></div>',
    '<div style={{html-safe (concat "background-image: url(" url ")")}}></div>',
  ],

  bad: [
    ...[
      'style="width: 100px"',
      'style="{{foo}} {{bar}}"',
      'style',
      'style=""',
      'style="color:blue;margin-left:30px;"',
    ].map(mapBadStyleExampleToTestRule),
    mapBadStyleExampleToTestRuleWithConfig('style={{foo}}', { allowDynamicStyles: false }),
  ],
});
