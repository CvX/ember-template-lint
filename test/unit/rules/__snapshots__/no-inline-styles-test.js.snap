// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-inline-styles <div style=""></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 13,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style=\\"\\"",
  },
]
`;

exports[`no-inline-styles <div style="{{foo}} {{bar}}"></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 28,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style=\\"{{foo}} {{bar}}\\"",
  },
]
`;

exports[`no-inline-styles <div style="color:blue;margin-left:30px;"></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 41,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style=\\"color:blue;margin-left:30px;\\"",
  },
]
`;

exports[`no-inline-styles <div style="width: 100px"></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 25,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style=\\"width: 100px\\"",
  },
]
`;

exports[`no-inline-styles <div style={{foo}}></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 18,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style={{foo}}",
  },
]
`;

exports[`no-inline-styles <div style></div>: logs errors 1`] = `
Array [
  Object {
    "column": 5,
    "endColumn": 10,
    "endLine": 1,
    "filePath": "layout.hbs",
    "line": 1,
    "message": "elements cannot have inline styles",
    "rule": "no-inline-styles",
    "severity": 2,
    "source": "style",
  },
]
`;
